lib/phoenix/socket.ex

```diff
@@ -15,8 +16,56 @@ defmodule Phoenix.Socket do
   * `topic` - The string topic, ie `"rooms:123"`
   * `transport` - The socket's transport, ie: `Phoenix.Transports.WebSocket`
   * `transport_pid` - The pid of the socket's transport process
+
+  ## Channels
+
+  Channels allow you to route pubsub events to channel handlers in your application.
+  By default, Phoenix supports both WebSocket and LongPoller transports.
+  See the `Phoenix.Channel.Transport` documentation for more information on writing
+  your own transports. Channels are defined within a socket handler, using the
+  `channel/2` macro, as seen below.
+
+  ## Socket Behaviour
+
+  Socket handlers are mounted in Endpoints and must define two callbacks:
+
+    * `connect/2` - receives the socket params and authenticates the connection.
+                    Often used to wire up default `%Phoenix.Socket{}` assigns
+                    for all channels.
+    * `id/1` - receives the socket returned by `connect/2`, and returns the
+               string id of this connection. Used for forcing a disconnect for
```

Comment:
It should be:

      * `id/1` - receives the socket returned by `connect/2`, and returns the
        string id of this connection. Used for forcing a disconnect for

Otherwise it shows all messed up in markdown.